### PR TITLE
fixes #822

### LINKED ISSUE #822: Move ability_upgrades to Redis
- [x] In insertMatch, store ability_upgrades in Redis as [[ability_id, ability_id...25]...10] to reduce space (TTL 3 days)
- [x] Look up ability_upgrades data when viewing a match page and add to match object
- [x] edit sql schema
- [x] add message explaining why we don't have ability_upgrades data for old matches
- [ ] alter table player_matches drop column ability_upgrades;

### PR DESCRIPTION
**Pull request by howardchung**
@albertcui 
- [ ] alter table player_matches drop column ability_upgrades;

### CODE REVIEW

File: queries.js [hunk 1]
```diff
@@ -93,21 +93,44 @@ function insertMatch(db, redis, queue, match, options, cb)
...
+    {
+        var ability_upgrades = {};
+        players.forEach(function(p)
+        {
+            ability_upgrades[p.player_slot] = p.ability_upgrades ? p.ability_upgrades.map(function(au)
+            {
+                return au.ability;
+            }) : null;
+        });
+        redis.setex("ability_upgrades:" + match.match_id, 60 * 60 * 24 * 7, JSON.stringify(ability_upgrades));
```
Comment (albertcui) [inline queries.js]: We might want to make this a config variable

**Comment by albertcui** [comment]
Mergeable after blog post is signed off.

File: _posts/ability-upgrade-removal.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+{{{
```
Comment (howardchung) [inline _posts/ability-upgrade-removal.md]: I'm not really sure this merits a blog post.

File: _posts/ability-upgrade-removal.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  "title": "Ability Upgrade Removal",
+  "date": "1-11-2016",
+  "author": "Albert"
+}}}
+
+### The Change
+
+We're always trying to operate YASP as financially efficient as possible, and today, we're making a change to ability upgrades
```
Comment (howardchung) [inline _posts/ability-upgrade-removal.md]: just say efficiently

File: _posts/ability-upgrade-removal.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+to do just that.
+
+**From now on, ability upgrades for new matches will be viewable for 24 hours after we add the match to our database. Afterwards,
```
Comment (howardchung) [inline _posts/ability-upgrade-removal.md]: 24 hours?  I have it set to 1 week atm

File: _posts/ability-upgrade-removal.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+we have to spend more on hard drive space each month.
+
+We currently have 100 million matches in our database. Each match's ability upgrade information takes 9 KB of data.
+In our opinion, this isn't entirely useful information a few hours after a match and especially between patches when
+heros are reworked. We're also not currently doing any aggregations over ability data (which would be very expensive).
+By cutting these from our database, we're saving a huge chunk of space.
+
+### And that means...
+
+We're going to use this space to improve the Y ASP experience. We're now going to go back and add all the Dota 2 matches ever played.
```
Comment (howardchung) [inline _posts/ability-upgrade-removal.md]: random space

File: _posts/ability-upgrade-removal.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+(That is, retrievable from the WebAPI.)
```
Comment (howardchung) [inline _posts/ability-upgrade-removal.md]: I don't know if we should mention this here.  We're not committing to it yet and we can announce it when we do so

File: queries.js (hunk 1 above)
Comment (howardchung) [inline queries.js]: Can refactor later if we need it to be configurable?

**Comment by albertcui** [comment]
Added MMR blog post. Merge when ready to deploy.

**Pull request merged**